main_test.go

```diff
@@ -162,3 +162,91 @@ func TestAddPath(t *testing.T) {
 	expected := []string{"existing", "new"}
 	assert.Equal(t, expected, addPath(paths, "new"))
 }
+
+func TestLinterFlags(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	app := kingpin.New("test-app", "")
+	setupFlags(app)
+	_, err := app.Parse([]string{"--linter", "a:b:c"})
+	require.NoError(t, err)
+	linter, ok := config.Linters["a"]
+	assert.True(t, ok)
+	assert.Equal(t, "b", linter.Command)
+	assert.Equal(t, "c", linter.Pattern)
+}
+
+func TestHistoricalLinterConfig(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	tmpfile, err := ioutil.TempFile("", "test-config")
+	require.NoError(t, err)
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write([]byte(`{"Linters": {"linter": "command:path"} }`))
+	require.NoError(t, err)
+	require.NoError(t, tmpfile.Close())
+
+	app := kingpin.New("test-app", "")
+	setupFlags(app)
+
+	_, err = app.Parse([]string{"--config", tmpfile.Name()})
+	require.NoError(t, err)
+	linter, ok := config.Linters["linter"]
+	assert.True(t, ok)
+	assert.Equal(t, "command", linter.Command)
+	assert.Equal(t, "path", linter.Pattern)
+}
+
+func TestMapLinterConfig(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	tmpfile, err := ioutil.TempFile("", "test-config")
+	require.NoError(t, err)
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write([]byte(`{"Linters":
+		{"linter":
+			{ "Command": "command" }}}`))
+	require.NoError(t, err)
+	require.NoError(t, tmpfile.Close())
+
+	app := kingpin.New("test-app", "")
+	setupFlags(app)
+
+	_, err = app.Parse([]string{"--config", tmpfile.Name()})
+	require.NoError(t, err)
+	linter, ok := config.Linters["linter"]
+	assert.True(t, ok)
+	assert.Equal(t, "command", linter.Command)
+	assert.Equal(t, "", linter.Pattern)
+}
+
+func TestMapAndCliLinterConfig(t *testing.T) {
+	originalConfig := *config
+	defer func() { config = &originalConfig }()
+
+	tmpfile, err := ioutil.TempFile("", "test-config")
+	require.NoError(t, err)
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write([]byte(`{"Linters":
+		{"linter": { "Command": "command" }}}`))
```

Comment:
Please make the value here different from `command` so the test also covers the override. Right now we don't know if the flag is overriding because they have the same value.